### PR TITLE
close #3: follow redirects

### LINKED ISSUE #3: Unable to download artefacts
The module drops a error when i download a artefact:

A example call:
```
githubreleases_download { '/tmp/gitui-linux-musl.tar.gz':
  author            => 'extrawurst',
  repository        => 'gitui',
  asset             => true,
  asset_filepattern => 'gitui-linux-musl.tar.gz',
}
```

The error message:
```
Debug: Checking asset gitui-linux-musl.tar.gz for RegExp (?i-mx:gitui-linux-musl.tar.gz)
Debug: Checking content type application/gzip for RegExp (?i-mx:.*)
Debug: Both are matching. Returning URL https://github.com/extrawurst/gitui/releases/download/v0.24.3/gitui-linux-musl.tar.gz
Error: Could not set 'present' on ensure: No such file or directory @ rb_sysopen - https://github.com/extrawurst/gitui/releases/download/v0.24.3/gitui-linux-musl.tar.gz (file: /etc/puppet/modules/ubuntudesktop/manifests/aspect/software.pp, line: 318)
Error: Could not set 'present' on ensure: No such file or directory @ rb_sysopen - https://github.com/extrawurst/gitui/releases/download/v0.24.3/gitui-linux-musl.tar.gz (file: /etc/puppet/modules/ubuntudesktop/manifests/aspect/software.pp, line: 318)
Wrapped exception:
No such file or directory @ rb_sysopen - https://github.com/extrawurst/gitui/releases/download/v0.24.3/gitui-linux-musl.tar.gz
Error: /Stage[main]/Ubuntudesktop::Aspect::Software/Githubreleases_download[/tmp/gitui-linux-musl.tar.gz]/ensure: change from 'absent' to 'present' failed: Could not set 'present' on ensure: No such file or directory @ rb_sysopen - https://github.com/extrawurst/gitui/releases/download/v0.24.3/gitui-linux-musl.tar.gz (file: /etc/puppet/modules/ubuntudesktop/manifests/aspect/software.pp, line: 318)
```

Github sends a 302 location header for the calculated uri.
The module needs to follow this location redirect.

### PR DESCRIPTION
**Pull request by scoopex**
This fixes #3

### CODE REVIEW

File: lib/puppet/provider/githubreleases_download/githubreleases_download.rb [hunk 1]
```diff
@@ -6,8 +6,19 @@
 
   def create
     download_url = get_download_url(@resource[:author], @resource[:repository], @resource)
-    download = open(download_url)
-    IO.copy_stream(download, @resource[:target])
+    uri = URI.parse(download_url)
+
+    # Follow redirects
+    response = Net::HTTP.get_response(uri)
+    while response.code.to_i == 302  # Redirect status code
```
Comment (dploeger) [inline lib/puppet/provider/githubreleases_download/githubreleases_download.rb]: Maybe add 301 as well here.

**Review by dploeger — CHANGES_REQUESTED** [review]
Awesome! Thanks a lot. Just a small remark.

Could you also please use a [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) style commit message?

File: lib/puppet/provider/githubreleases_download/githubreleases_download.rb (hunk 1 above)
Comment (scoopex) [reply lib/puppet/provider/githubreleases_download/githubreleases_download.rb]: Thanks for the quick review! Changed the check to check both codes.

**Comment by scoopex** [comment]
> Awesome! Thanks a lot. Just a small remark.
> 
> Could you also please use a [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) style commit message?

Done

**Review by dploeger — APPROVED** [review]
Awesome. Thanks a lot! 🎁

**Pull request merged**